### ISSUE TITLE
Is it possible to have a high accuracy with a high loss?

### ISSUE THREAD

**Issue by zmykevin** [BVLC/caffe#4520]
Recently I am implement a neural network to conduct a classification task, where I just need to classify the datasets images into two different categories. I used lenet net as a start up structure, and train the network for 10000 iterations. I observed that the test accuracy increase from the initial 0.57 to 0.91 just after 500 iterations of training, but still have a pretty high loss as 0.47. After 10000 iterations the accuracy is about 0.95, and the loss is about 0.1. There are about 8000 test images. After I train the network, I write a simple program to check whether the prediction for each test image is the same as the ground truth, and actually find about 1900 failure prediction, which is almost 25% of the test image dataset. This is weird to me. I double checked the accuracy layer, and the implementation of my testing code, I am pretty sure my code is doing right. Therefore, how could this happen that I reached 95% accuracy with a set of random batch sampled from the testing cases, which only get 75% accuracy with the full testing cases? In addition, I think the loss is a little too high for such a high accuracy. Can anyone give some intuition of what happens and how could I resolve it?

**Comment by Apple-Of-Eden** [comment]
Hi zymkevin,

So let me first say that it's not _technically_ impossible to have a high loss and a high accuracy. It depends. For instance, if your validation or testing dataset has 90% of the images as one class A (assuming there's a maximum of two classes, A and B, you're using), then it's possible to get a 90% accuracy if you train incorrectly and always guess A. However, in that example, the CNN isn't exactly going to do well in real world classification tasks as it only guesses one class...

That being said, generally speaking, a lower loss means a higher accuracy. Your loss of 0.47 doesn't seem unreasonable for an accuracy of 95% given that you're doing binary classification. I'm coincidentally doing binary classification myself and have a loss of around 0.40 and an accuracy of around 90%. 

But the question remains, why are you only getting 75% accuracy with your test data? Well, I'm not sure. It's possible that you're over-fitting, but even so I wouldn't think that you'd have a difference of 75% vs. 95%! Do you stop training as soon as accuracy no longer improves?

Do you know what images are confidently being guessed incorrectly? Some pattern there? 

Hopefully, this isn't the case, but you're sure that your validation images are distinct from your training images and that your testing images are also distinct from your training images?

As an aside, I think this usage question is supposed to be asked in the google group forums. Here: https://groups.google.com/forum/#!forum/caffe-users

**Comment by zmykevin** [comment]
Hey Eden, thank you very much for the thoughtful reply! I think I have a better understanding of the relationship between loss and accuracy right now. I didn't have validation set of data, just train and test. When I train the data, I validate it with the testing samples, and during the process, I found that the accuracy will go up to 95% ultimately, and the loss will drop from 0.6 to 0.1. I didn't stop training after the accuracy stops increaing, but as I set the snap shot step smaller, I actually keep the trained parameters every 1000 iterations. I do notice that when I use the parameters trained after it reaches the maximum accuracy, the error cases will drop when I use those parameters to test. The accuracy is around 87%, which is still far below 95%.  I do have a rough look over the failure case, and I guess I have a sense that the model is not able to handle those cases because some of them are really in between the two binary cases. The input to my network is just the raw pixel values of images, but as I am dealing with classification based more on color chracteristics, I will try to add a hue channel to the input, and also modify the network a little more complex to see if it will help. 
I really appreciate for the answer you give me, and also the suggestion about the google forums. I am looking forward for more interesting discussion with people like you!.

**Comment by seanbell** [comment]
Please ask modeling questions on the mailing list, thanks!

From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:
> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
